crates/orchestrator/src/database/mongodb/mod.rs

```diff
@@ -290,4 +294,34 @@ impl Database for MongoDb {
 
         Ok(results)
     }
+
+    async fn get_jobs_by_status(&self, job_status: JobStatus, limit: Option<i64>) -> Result<Vec<JobItem>> {
+        let filter = doc! {
+            "job_status": bson::to_bson(&job_status)?
+        };
+
+        let mut find_options = None;
+        if let Some(val) = limit {
+            find_options = Some(FindOptions::builder().limit(Some(val)).build())
+        };
+
+        let mut jobs = self
+            .get_job_collection()
+            .find(filter, find_options)
+            .await
+            .expect("Failed to fetch jobs by given job type and status");
+
+        let mut results = Vec::new();
+
+        while let Some(result) = jobs.next().await {
+            match result {
+                Ok(job_item) => {
+                    results.push(job_item);
+                }
+                Err(e) => return Err(e.into()),
+            }
+        }
+
+        Ok(results)
```

Comment:
```rust
        let jobs: Vec<JobItem> = self
            .get_job_collection()
            .find(filter, find_options)
            .await
            .expect("Failed to fetch jobs by given job type and status")
            .try_collect()
            .await?;

        Ok(jobs)
```

this looks like it would do the job (took help from Claude :)). if it works, we can use it above as well